cmake/Modules/FindClangFormat.cmake

```diff
@@ -0,0 +1,94 @@
+#
+# .rst: FindClangFormat
+# ---------------
+#
+# The module defines the following variables
+#
+# ``CLANG_FORMAT_EXECUTABLE`` Path to clang-format executable
+# ``CLANG_FORMAT_FOUND`` True if the clang-format executable was found.
+# ``CLANG_FORMAT_VERSION`` The version of clang-format found
+# ``CLANG_FORMAT_VERSION_MAJOR`` The clang-format major version if specified, 0
+# otherwise ``CLANG_FORMAT_VERSION_MINOR`` The clang-format minor version if
+# specified, 0 otherwise ``CLANG_FORMAT_VERSION_PATCH`` The clang-format patch
+# version if specified, 0 otherwise ``CLANG_FORMAT_VERSION_COUNT`` Number of
+# version components reported by clang-format
+#
+# Example usage:
+#
+# .. code-block:: cmake
+#
+# find_package(ClangFormat) if(CLANG_FORMAT_FOUND) message("clang-format
+# executable found: ${CLANG_FORMAT_EXECUTABLE}\n" "version:
+# ${CLANG_FORMAT_VERSION}") endif()
+
+find_program(CLANG_FORMAT_EXECUTABLE
+             NAMES
+                   clang-format-9
+                   clang-format-9.0
+                   clang-format-8
+                   clang-format-8.0
+                   clang-format-7
+                   clang-format-7.0
+                   clang-format-6.0
+                   clang-format-5.0
+                   clang-format-4.0
+                   clang-format-3.9
+                   clang-format-3.8
+                   clang-format-3.7
+                   clang-format-3.6
+                   clang-format-3.5
+                   clang-format-3.4
+                   clang-format-3.3
+                   clang-format  # least priority
+             DOC "clang-format executable")
+mark_as_advanced(CLANG_FORMAT_EXECUTABLE)
+
+if(CLANG_FORMAT_EXECUTABLE)
```

Comment:
It's conventional to end finder scripts with the following:

```cmake
include(FindPackageHandleStandardArgs)
find_package_handle_standard_args(
  ClangFormat
  REQUIRED_VARS
    CLANG_FORMAT_EXECUTABLE
    CLANG_FORMAT_VERSION
)
```

This, among other things, will set the `CLANG_FORMAT_FOUND` flag. Also print a report, handle `QUIET` and `REQUIRED` flags of `find_package` and so on.

Please put this in the end of the file and remove this early-ending block.

Reply:
Added this along with VERSION_VAR for version checking. MUCH BETTER now. Didn't know this trick.